libs/providers/config-cat-web/src/lib/config-cat-web-provider.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  EvaluationContext,
+  FlagNotFoundError,
+  JsonValue,
+  OpenFeatureEventEmitter,
+  Paradigm,
+  ParseError,
+  Provider,
+  ProviderEvents,
+  ProviderNotReadyError,
+  ResolutionDetails,
+} from '@openfeature/web-sdk';
+import { TypeMismatchError } from '@openfeature/core';
+import {
+  isType,
+  PrimitiveType,
+  PrimitiveTypeName,
+  toResolutionDetails,
+  transformContext,
+} from '@openfeature/config-cat-core';
+import { getClient, IConfig, IConfigCatClient, IConfigCatClientSnapshot, PollingMode } from 'configcat-js-ssr';
+
+export class ConfigCatWebProvider implements Provider {
+  public readonly events = new OpenFeatureEventEmitter();
+  private readonly _clientParameters: Parameters<typeof getClient>;
+  private _client?: IConfigCatClient;
+  private _clientSnapshot?: IConfigCatClientSnapshot;
+
+  public runsOn: Paradigm = 'client';
+
+  public metadata = {
+    name: ConfigCatWebProvider.name,
+  };
+
+  constructor(...params: Parameters<typeof getClient>) {
+    this._clientParameters = params;
+  }
+
+  public static create(...params: Parameters<typeof getClient>) {
+    return new ConfigCatWebProvider(...params);
+  }
+
+  public async initialize(): Promise<void> {
+    const originalParameters = this._clientParameters;
+    originalParameters[2] ??= {};
+
+    const pollingMode = originalParameters[1];
+    const options = originalParameters[2];
+    const oldSetupHooks = options.setupHooks;
+
+    options.setupHooks = (hooks) => {
+      oldSetupHooks?.(hooks);
+
+      hooks.on('clientReady', () => {
+        this.events.emit(ProviderEvents.Ready);
+      });
+
+      hooks.on('configChanged', (projectConfig: IConfig | undefined) => {
+        this.events.emit(ProviderEvents.ConfigurationChanged, {
+          flagsChanged: projectConfig ? Object.keys(projectConfig.settings) : undefined,
+        });
+
+        if (this._client) {
+          this._clientSnapshot = this._client.snapshot();
+        }
+      });
+
+      hooks.on('clientError', (message: string, error) => {
+        this.events.emit(ProviderEvents.Error, {
+          message: message,
+          metadata: error,
+        });
+      });
+    };
+
+    const client = getClient(...originalParameters);
+
+    if (pollingMode !== PollingMode.AutoPoll) {
```

Comment:
I don't think we should allow polling modes other than Auto Polling. As the docs point out:

> Please note that when you create and utilize a snapshot, it won't refresh your local cache once the cached config data expires. Additionally, when working with [shared caching](https://configcat.com/docs/advanced/caching/#shared-cache), creating a snapshot also doesn't trigger a sync with the external cache, since the snapshot only captures the config instance stored in the client's memory. Therefore, it's recommended to use snapshots in conjunction with the Auto Polling mode, where the SDK automatically updates the local cache in the background. For other polling modes, you'll need to manually initiate a cache refresh by invoking forceRefreshAsync.

That is, to make snapshots actually work with Manual Polling or Lazy Loading, we'd need to implement a background job that keeps the SDK's cache up to date. IMO that wouldn't make much sense as it would be pretty much a re-implementation of what Auto Polling already does.

So we should either throw an exception or ignore the polling mode option if it's not set to Auto Polling. Probably throwing an exception would be the more transparent behavior to the user.

Reply:
Makes sense, will update that later or tomorrow @adams85 :)